site/app/controllers/student/SubmissionController.php

```diff
@@ -360,7 +360,7 @@ public function ajaxBulkUpload($gradeable_id) {
             return $this->uploadResult("Invalid gradeable id '{$gradeable_id}'", false);
         }
 
-        $num_pages = $_POST['num_pages'];
+        $num_pages = rawurlencode($_POST['num_pages']);
```

Comment:
Why is this necessary?

Reply:
I added the rawurlencode as per the suggestion of @ziesski. The reason for adding it was that URL encoding is a good practice to handle unsafe characters.﻿